ClosedXML.Tests/Excel/CalcEngine/FunctionsTests.cs

```diff
@@ -249,6 +249,57 @@ public void MMult()
             Assert.AreEqual(102.0, actual);
         }
 
+        [Test]
+        public void MMult_HandlesNonSquareMatrices()
+        {
+            IXLWorksheet ws = new XLWorkbook().AddWorksheet("Sheet1");
+
+            // 2x3
+            ws.Cell("A1").SetValue(1).CellRight().SetValue(3).CellRight().SetValue(5);
+            ws.Cell("A2").SetValue(2).CellRight().SetValue(4).CellRight().SetValue(6);
+
+            // 3x4
+            ws.Cell("A3").SetValue(10).CellRight().SetValue(13).CellRight().SetValue(16).CellRight().SetValue(19);
+            ws.Cell("A4").SetValue(11).CellRight().SetValue(14).CellRight().SetValue(17).CellRight().SetValue(20);
+            ws.Cell("A5").SetValue(12).CellRight().SetValue(15).CellRight().SetValue(18).CellRight().SetValue(21);
+
+            Object actual;
+
+            // 2x4 output expected:
+            // 103, 130, 157, 184
+            // 136, 172, 208, 244
+            ws.Cell("A6").FormulaA1 = "MMult(A1:C2, A3:D5)";
+
+            actual = ws.Cell("A6").Value;
+            Assert.AreEqual(103.0, actual);
+
+            ws.Cell("A7").FormulaA1 = "Sum(MMult(A1:C2, A3:D5))";
+            actual = ws.Cell("A7").Value;
+
+            Assert.AreEqual(1334, actual);
+        }
+
+        [TestCase("A2:C2", "A3:C3")] // 1x3 and 1x3
+        [TestCase("A2:C4", "A5:C5")] // 3x3 and 1x3
+        [TestCase("A2:C5", "A6:D6")] // 3x4 and 1x4
+        public void MMult_ThrowsWhenArray1RowsNotEqualToArray2Cols(string array1Range, string array2Range)
+        {
+            IXLWorksheet ws = new XLWorkbook().AddWorksheet("Sheet1");
+
+            ws.Cells($"{array1Range}").Value = 1.0;
+            ws.Cells($"{array2Range}").Value = 1.0;
+
+            ws.Cell("A1").FormulaA1 = $"MMULT({array1Range},{array2Range})";
+
+            var error = Assert.Throws<ArgumentException>(() =>
```

Comment:
Please refer to https://support.microsoft.com/en-us/office/mmult-function-40593ed7-a3cd-4b6b-b9a3-e4ad3c7245eb?ns=excel&version=19&syslcid=1033&uilcid=1033&appver=zxl190&helpid=xlmain11.chm60213&ui=en-us&rs=en-us&ad=us

The correct exception to be thrown is `CellValueException`. I see this is a legacy issue, but please add a commit to fix this.

Bonus points if you can also add a test for when any cells are empty or contain text.

Reply:
Yes, please. That sounds perfect.

Reply:
I've switched the `ArgumentException` for size mismatches to `CellValueException`.

I've also added the `CellValueException` wrapper for format exceptions emerging from `MathTrig.GetArray`, and a test covering a few cases.